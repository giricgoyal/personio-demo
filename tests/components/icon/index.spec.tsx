import React from 'react'
import Icon from 'src/components/icon'
import { render } from 'tests/test-utils'

describe('components/icon', () => {
    let container, rerender

    beforeEach(() => {
        ;({ container, rerender } = render(<Icon icon="filter" size="small" />))
    })

    test('should render and match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('When invalid icon is provided', () => {
        beforeEach(() => {
            rerender(<Icon icon="invalid" />)
        })

        test('should not render and match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })

    describe('When spin is provided', () => {
        beforeEach(() => {
            rerender(<Icon icon="spinner" spin />)
        })

        test('should render and match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })
})
