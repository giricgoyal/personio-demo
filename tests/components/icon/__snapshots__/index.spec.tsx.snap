// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/icon When invalid icon is provided should not render and match snapshot 1`] = `<div />`;

exports[`components/icon When spin is provided should render and match snapshot 1`] = `
<div>
  <i
    class="fas fa-spinner  fa-spin"
  />
</div>
`;

exports[`components/icon should render and match snapshot 1`] = `
<div>
  <i
    class="fas fa-filter "
  />
</div>
`;
